roles/lustre_client/tasks/install.yml

```diff
@@ -0,0 +1,100 @@
+---
+- name: Install packages for build environment to support Dynamic Kernel Module Support (DKMS).
+  yum:
+    state: latest
+    update_cache: true
+    name:
+      - asciidoc
+      - audit-libs-devel
+      - automake
+      - bc
+      - binutils-devel
+      - bison
+      - device-mapper-devel
+      - elfutils-devel
+      - elfutils-libelf-devel
+      - expect
+      - flex
+      - gcc
+      - gcc-c++
+      - git
+      - glib2
+      - glib2-devel
+      - hmaccalc
+      - keyutils-libs-devel
+      - krb5-devel
+      - ksh
+      - libattr-devel
+      - libblkid-devel
+      - libselinux-devel
+      - libtool
+      - libuuid-devel
+      - libyaml-devel
+      - lsscsi
+      - make
+      - ncurses-devel
+      - net-snmp-devel
+      - net-tools
+      - newt-devel
+      - numactl-devel
+      - parted
+      - patchutils
+      - pciutils-devel
+      - perl-ExtUtils-Embed
+      - pesign
+      - python-devel
+      - redhat-rpm-config
+      - rpm-build
+      - systemd-devel
+      - tcl
+      - tcl-devel
+      - tk
+      - tk-devel
+      - wget
+      - xmlto
+      - yum-utils
+      - zlib-devel
+  become: true
+
+- name: Install kernel tools/extras required for compiling kernel modules.
+  yum:
+    state: latest
+    update_cache: true
+    name:
+      - kernel
+      - kernel-devel
```

Comment:
Maybe a bit of an overkill to install the kernel source everywhere, but well, i'm fine with it.

Reply:
For the record: this is required when using DKMS as an updated Lustre module will be compiled from scratch on each and every machine that needs a Lustre client. The alternative would be to use a pre-compiled Lustre client, package that as RPM and add it to a custom repo, which is what we have now in the SpaceWalk channel, but that is not the DKMS way....